### DISCUSSION TITLE
Main

### DISCUSSION THREAD

**Discussion by comradekingu** [WeblateOrg/weblate#5753] [General] [answered]
https://weblate.org/nb/news/archive/respect-support-and-privacy/

And thus, nobody mentioned "~~slave~~" or "~~master~~" ever again, and everyone forget their made up reasons for not doing so.
No time was laid to waste discussing a meaningless initiative, as everyone bent over at the hip sideways, praising the corporate world as having done something empowering in lowering the mental ceiling. The new "master" naming. A marvel, mind.

There is no actual reason to be upset about its etymology.
Foregoing that insight, this newly found exclusive logic _creates_ a problem that is not going away in terms of default branch naming.
A level of foresight that seems a rational default in terms of accountability, the very thing that gives people agency in their own right.

>When I was a little boy, I had a master and slave. Luckily only inside the computer. I didn’t understand in the start why some parts were named that way. Many years after, we are talking about these things all around the world. And it’s good that we are. 

I take this to mean `primary master` and `secondary master` P-ATA IDE cables, and the equivalent `primary slave` and `secondary slave` ones. The issue, for those even older, was the binary issue of figuring out which way to put them in.
That issue was solved. Few people remember, and it is a good thing they don't.
Things used to be difficult, and slavery used to be a thing.
It is however sad some people should come to not only forget, but also misrepresent adjacent use of technology
to undermine history. It isn't doing much for people living in slavery today, or in the very least not in excess, which is to say it is just particularly distasteful in doing nothing.

The master/slave naming scheme has been the default since at least subversion.
For its use in technology, there is absolutely no reason to make this something to get hung up about,
because that problem wasn't even part of the equation there.
Treating people selectively as non-adults is, counterproductive.

In my day respect wasn't just given, and I still treat people to this hard-earnt luxury. I respect myself
by using my critical faculties? Cookie-tracking fall out of fashion when fingerprinting has taken its place. Interesting.
I celebrate less bandwidth wasted?
There is no fanfare in loading stats.wp.com in a blogpost about not using cookies, for a site that requires logging in to do anything…
Oh but they are the good guys. Yes. "Main", suspect.

Whose interests are really at play here, and who benefits? GitHub has some downtime, who cares.
Proprietary service owned by monopoly abuser making proprietary solutions holding back technology decades,
with insight into branches named "private", using the non-proprietary Git. Is the name "Git" the issue there?

I suggest instead a cursory introduction to slavery as it exists, which is to say completely outside of some minorities
failing to integrate into the US. Like data-mining in the west, it is flourishing in Africa, in the traditional sense. Chattel slavery.
All solved by not calling it third-world like the people that live there. Developing countries, or as the global bank puts it
"Global South".
For case of argument, naming branches "chattelslavery" would at least seem like a honest attempt at bringing a real problem to the fore.
Can't talk about it? Don't care. Default branch naming not the place? I agree.

Naming branches "main" isn't slacktivism in the sense that doing more along the lines of what it
is would amount to activism. It is diametrically the opposite, because in being the imposed default, it is as passive as can be.
There are actual ways to do actual things in libre software that don't amount to putting meaningless letters onto things.
Nor is "main" neccisarily the main branch used, but we digress.

Instead this non-virtuous, nonsensical thinking of convenience falls flat of solving real issues.
The idea is that specifically some minorities that weren't contributing on equal footing to
libre software, will all of a sudden do so now. Because of an invented focus is made specifically
"for" them, at a level it is impossible for anyone contributing to avoid.
That is an awful idea of injecting irrelevancy where it need not be (and neither is) relevant.
Happily one can report a diversity of minorities doing particularly well in contributing.
Before and after. Of course. The real tell here is nobody expected anything else.

The same awful idea of presenting immutable characteristics as a reason why inclusion
might not happen, was a recently imposed idea presented at the start of contributor guidelines,
in the form of the first lines of a CoC.

That is luckily failing, in similar fashion to its creator going on to champion non-libre software
licensing, supposedly getting rid of biases. One will find it does nothing but add
biases, and mutual exclusivity, at a fundamental level. There is a pattern there.
Maybe the corporate policy language-only of they-are-all-the-same conglomerates
isn't really trying to upend or change anything?

In its wisdom, Weblate already got rid of this virtue-signalling, by dropping said CoC
that was introduced in no part of its making. Its community now stands 30k strong
on the host platform alone. Increased growth, no issues.

The latest blogpost had me in despair, hot on the heels of me trying to get stuff like
", so that's another handy you get with Weblate. " out of there, for perhaps
a different take on professionalism than the current US unculture.
Aping corporate anti-humanity and anti-community obviously has no place in libre software.
It is as fruitless and warrant-less as its tautology of taking credit for people being nice,
suggesting it will do so, or imposing authority where none needs be instated.

Libre software is actually the closest thing to a meritocracy we have.
Equal opportunity, not equity. I will freely and happily admit I am among my superiors here.
Some people do one thing well, others can have another idea or do something else.
It is all ready for the taking.
Every building block is already there, as with the cost of entry continually going in the
direction of abundance. I find there is enough difficulty in using Git as it is.

If the premise is that little bit of convenience away from actually making sense,
and works on the premise of "doing something", under the guise of
"getting somewhere", when that is some-place else entirely, I suggest thinking first.
What does it actually do, fix anything?

To the contrary it makes contributing that little bit more difficult.

At least there was an invite to discussion, so all hope is not lost.
To turn the tables on implicit bias, would Slavic people, (Slavs) think branch naming is
a hindrance of anyone's kind to technology?

One would hope not. Far removed from US politics, and even under its rule, they have
been instrumental in technology since its inception, much in part a field of their own creation.
That is how you make the point of inclusion.

**Comment by nijel** [comment, marked as answer]
We're not going to change the branch name back. Even if you don't acknowledge other reasons, there is one purely technical - the default branch for newly created repos is `main` and it's good to be consistent within the project. For several months we had a mixture of `main` and `master` and that caused technical problems (for example it blocked sharing some CI pipelines).

> The master/slave naming scheme has been the default since at least subversion.

There was no master/slave in subversion, the master branch was introduced in Git. It was chosen by Petr Baudiš and this is his current opinion on that: https://twitter.com/xpasky/status/1271477451756056577. There is also a more in-depth interview with him on this topic available in Czech here: https://www.root.cz/clanky/cechum-neni-vymena-slova-master-v-gitu-srozumitelna-rika-petr-pasky-baudis/

**Reply by comradekingu** [reply]
Maybe so, I just remembered it being that. Even if it isn't even from Bitkeeper, 16 years within what is now the most commonly used VCS is a fair bit of established technical debt.

He got it right even if he can't explain why. Similarly people think they are doing something they aren't in changing it, and can't explain why. Just because GitHub changes its scheme and it creates inconsistencies, doesn't mean they have to be corrected into whatever scheme that is. It lends no credibility to the reasoning behind why the old scheme was bad.

The imposition here is that it creates a rationale.

>If we have even a slight chance to make the world kinder, better, more inclusive, and less offensive, let’s do it.

To just deal with the "offensive" there. If I grant master, or slave is offensive, by the same logic that will have created _more_ offence, given the multitude of fields that employ this wording, by that fact alone.

It becomes impossible to talk about technology, electronics, philosophy, etc. in a manner that makes sense around people like this, and that is why people don't. Yet another exercise in walking on eggshells that only do harm to people that could be in a better mental state.

In turn, an artificial need to change other systems is created, because as far as prescriptive language without being language goes, logic is out.

It creates problems, and it doesn't solve any. Just lost engineering time in a world that needs engineering solutions.
We don't need solutions engineered into a pretend world.

Not to mention you have to double-think yourself into a world where you pick the people it is offensive for. In itself dehumanizing. Nobody would ever do that to Slavic people, and they are Slavs, by name.
Curiously stuff like this doesn't fly at all there, and by accounts they enjoy inclusion rates in technology anyone could hope to model themselves on. They learnt German back in the day to get access to higher learning. They sure did seem more concerned with komitern than IDE cables or the order of characters in Unicode, etc. etc.

The better naming is more of a toss-up, but after the fact the better option clearly isn't to change anything, much less for purely non-technical reasons, and absolutely not when it means things are going to be inconsistent for the foreseeable future. Even simple guides for contributing to other projects fail without back-end trickery, and all the correct ones are now hit or miss. If you make your own, there is now a choice, with implications.

There is a lot of people who don't bow to things like this. Especially when it creates technical problems and isn't a sound decision to begin with.

Moreover, you can just select whatever you want https://github.com/WeblateOrg/Weblate/settings/branches
as your default on GitHub. I just changed mine back to "master" and created a repo to fix some blogposts. Any new repo I create will thus be "master".

How do you know you aren't changing the name? People might take offence for no reason again.

>If we have even a slight chance to make the world

Under the headline "chance", that is _not even wrong_…

I will acknowledge any and all good reasons why changing the default, or changing away from master is a good idea.

Edit: Read the link now. If the auto-translation is anything to go by, it is all over the place. Interesting though. Would be much less interesting for me to pick apart the logic in writing, and maybe not fair since I don't really know what it says.

**Reply by orangesunny** [reply]
@comradekingu 

We from the Weblate team will not spend hours in a conversation that is already solved by other people who are much more knowledgeable and much more involved in the issue. Also, I am not going to accept fighting strawmen or any other argumentative fouls or sense-deflectors.

Weblate statement is included in the blog post. We renamed the branches to show respect and support and also make things technically easier. You have the right to disagree, but you know the statement to respect.

Bringing up an erroneous interpretation of the part of Slavic nations’ history while you are talking to a person from the Slavic nation is wrong and unrelated to this topic.

If you want to understand the reasons behind renaming the git branches, it is possible; there are plenty of [sources](https://lore.kernel.org/git/nycvar.QRO.7.76.6.2006091126540.482@ZVAVAG-DN14RQO.ybpnyqbznva/) to start. If you want to show respect to other people, accepting the change is a good thing to do.

If the only thing you want is to publish your perspective on the world, we can’t follow.

**Reply by comradekingu** [reply]
I don't have the right to disagree here, no.

I wouldn't think developers would spend time on it either, but nijel actually had some productive input, wheras someone else in the Weblate team wrote:
>If you have any questions or ideas about this, let’s discuss them kindly in the community!

And then followed it up with

>fighting strawmen or any other argumentative fouls or sense-deflectors.

I have the outmost respect for people working with technology, advancing it beyond meaningless positions in it.
Also I respect language enough not to misrepresent it for the sake of supposed feelings to the contrary. Were I to do that, it would equally "limit" the same perpetually offended people from the fields I actually have my education in. Sadly they were nowhere to be found at the time, and aren't by their own metrics attempting to be now.

Regardless of me not placing myself in the history of my people even as a Norseman, that hasn't been the case historically speaking either.

Not treating some people differently on a fundamental level for immutable characteristics is actually easy. Representation in achievement is hard.
